src/python/zquantum/core/hamiltonian.py

```diff
@@ -0,0 +1,72 @@
+from openfermion.ops import QubitOperator
+from typing import Tuple, List
+
+
+def is_comeasureable(
+    term1: Tuple[Tuple[int, str], ...], term2: Tuple[Tuple[int, str], ...]
+) -> bool:
+    """Determine if two Pauli terms are co-measureable. Co-measureable means that
+       for each qubit: both terms apply the same Pauli operator, or at least one term
+        applies the identity.
+    Args:
+        term1: a product of Pauli operators represented in openfermion style
+        term2: a product of Pauli operators represented in openfermion style
+    Returns:
+        bool: True if the terms are co-measureable.
+    """
+
+    for op1 in term1:
+        for op2 in term2:
+
+            # Check if the two Pauli operators act on the same qubit
+            if op1[0] == op2[0]:
+
+                # Check if the two Pauli operators are different
+                if op1[1] != op2[1]:
+                    return False
```

Comment:
Yes this is the intended behavior. I changed the docstring to be more clear. The way it was originally worded was a bit confusing because we don't explicitly store identity operators in our representation of Pauli strings.